code/obj/item/gun/kinetic.dm

```diff
@@ -870,6 +880,294 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
+		AddComponent(/datum/component/holdertargeting/fullauto, 1.5, 1.5, 1)
+		..()
+
+	attack_self(mob/user)
+		..()	//burst shot has a slight spread.
+		if (istype(current_projectile, /datum/projectile/bullet/nine_mm_surplus/auto))
+			spread_angle = 10
+			shoot_delay = 4
+		else
+			spread_angle = 8
+			shoot_delay = 5
+
+	//warcrimes brought to you by bullets telling guns how to shoot!
+	attackby(obj/item/ammo/bullets/b, mob/user)
+		var/obj/previous_ammo = ammo
+		var/mode_was_auto = current_projectile.fullauto_valid
+		..()
+		if(previous_ammo.type != ammo.type)  // we switched ammo types
+			if(istype(ammo, /obj/item/ammo/bullets/nine_mm_surplus))
+				if(mode_was_auto)
+					set_current_projectile(new/datum/projectile/bullet/nine_mm_surplus/auto)
+					projectiles = list(new/datum/projectile/bullet/nine_mm_surplus/burst, current_projectile)
+				else
+					set_current_projectile(new/datum/projectile/bullet/nine_mm_surplus/burst)
+					projectiles = list(current_projectile, new/datum/projectile/bullet/nine_mm_surplus/auto)
+			else if(istype(ammo, /obj/item/ammo/bullets/bullet_9mm))
+				if(mode_was_auto)
+					set_current_projectile(new/datum/projectile/bullet/bullet_9mm/smg/auto)
+					projectiles = list(new/datum/projectile/bullet/bullet_9mm/smg, current_projectile)
+				else
+					set_current_projectile(new/datum/projectile/bullet/bullet_9mm/smg)
+					projectiles = list(current_projectile, new/datum/projectile/bullet/bullet_9mm/smg/auto)
+
+/obj/item/gun/kinetic/greasegun
+	name = "\improper Grease Gun"
+	desc = "A stamped metal SMG, cheaply produced years ago, and found in the weirdest places today."
+	icon_state = "grease"
+	item_state = "grease"
+	icon = 'icons/obj/items/guns/kinetic48x32.dmi'
+	spread_angle = 14
+	shoot_delay = 5
+	has_empty_state = TRUE
+	w_class = W_CLASS_SMALL
+	force = MELEE_DMG_PISTOL
+	ammo_cats = list(AMMO_SMG_9MM)
+	max_ammo_capacity = 30
+	auto_eject = TRUE
+	fire_animation = TRUE
+	default_magazine = /obj/item/ammo/bullets/nine_mm_surplus/mag_grease
+	var/grease = 0 //guh
+	get_desc(dist, mob/user)
+		if (grease == 0)
+			. += "It's all seized up and could do with maintenance."
+		else if (grease < 0)
+			. += "It's, er, all sticky and covered glue. WHY is it covered with glue???"
+		else
+			. += "It's greasy, alright..."
+
+	attack_self(mob/user as mob)
+		if(ishuman(user))
+			if(two_handed)
+				setTwoHanded(0) //Go 1-handed.
+				src.spread_angle = initial(src.spread_angle)
+				icon_state = "grease"
+			else
+				if(!setTwoHanded(1)) //Go 2-handed.
+					boutput(user, SPAN_ALERT("Can't switch to 2-handed while your other hand is full."))
+				else
+					icon_state = "greaseunfolded"
+					src.spread_angle = 6
+		..()
+
+	shoot() // fuck up firerate speed
+		var/datum/component/holdertargeting/fullauto/firemode = GetComponent(/datum/component/holdertargeting/fullauto)
+		var/delay = firemode.delaystart*10
+		if (grease > 0)
```

Comment:
Would be nice to have a message that plays when the gun "runs out" of grease, something like "The Grease Gun sputters and clanks" (I don't know what noise it would make but *something*)

Reply:
maybe. it being a brief speedup makes more sense than it being a distinct state IMO. and less chat clutter.